src/ast/passes/semantic_analyser.cpp

```diff
@@ -1914,6 +1915,127 @@ void SemanticAnalyser::visit(While &while_block)
   loop_depth_--;
 }
 
+void SemanticAnalyser::visit(For &f)
+{
+  bool error = false;
+
+  if (!bpftrace_.feature_->has_helper_for_each_map_elem()) {
+    LOG(ERROR, f.loc, err_)
+        << "Missing required kernel feature: for_each_map_elem";
```

Comment:
You mention the kernel version in the docs, want to also mention it here?

Reply:
Our feature detection logic means that this can run on older kernels if they've backported the required helper, so mentioning it in the error message could be misleading.

I wasn't sure about including it in the docs, but put it in because the docs for while loops mentioned it too. It's nice to have a rough idea of when a feature is available, but backporting happens in a few distros so having it in the docs could be misleading too... 🤷‍♂️

Reply:
I think it's ok to be verbose in these scenarios e.g. mention the kernel version but it may be specific to the distro if the changes were backported.